lib/aio/jsonutil.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (C) 2023 Red Hat, Inc.
```

Comment:
This is less about untrusted input and more about convincing mypy that we're not making mistakes.  That's how this stuff came about in the first place in the cockpit bridge as well (which also doesn't deal with untrusted inputs).

Reply:
Yeah, I get it in the bridge, but we don't need to be concerned about untrusted input here. I like mypy, but having to have such overhead for accessing a simple toml file feels too expensive to me.

Reply:
It's also for the job objects which we read off of the network.

Also keep in mind, the way that we define the job object, for example, these helpers actually help us get the types right in the first place.  Consider:

```
class Job:
    def __init__(self, obj: JsonObject) -> None:
        # test subject specification
        self.repo = get_str(obj, 'repo')
        self.sha = get_str(obj, 'sha', None)
        self.pull = get_int(obj, 'pull', None)
        self.branch = get_str(obj, 'branch', None)
        self.target = get_str(obj, 'target', None)

 ...etc...
```

mypy only knows the types of those attributes because of `get_str()` et al. being correctly typed.  This has found real issues during refactoring and in general has been a big help...

Reply:
OK, resolving..